contracts/poap/src/contract.rs

```diff
@@ -0,0 +1,451 @@
+use crate::error::ContractError;
+use crate::msg::{
+    ExecuteMsg, InstantiateMsg, QueryConfigResponse, QueryEventInfoResponse, QueryMsg,
+};
+use crate::state::{
+    Config, EventInfo, CONFIG, CW721_ADDRESS, EVENT_INFO, MINTER_ADDRESS, NEXT_POAP_ID,
+};
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_binary, Addr, Binary, CosmosMsg, Deps, DepsMut, Env, MessageInfo, Reply, ReplyOn, Response,
+    StdResult, SubMsg, Timestamp, WasmMsg,
+};
+use cw2::set_contract_version;
+use cw721_base::{
+    msg::ExecuteMsg as Cw721ExecuteMsg, InstantiateMsg as Cw721InstantiateMsg, MintMsg,
+};
+use cw_utils::parse_reply_instantiate_data;
+use url::Url;
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:poap";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+// actions consts
+const ACTION_ENABLE_MINT: &str = "enable mint";
+const ACTION_DISABLE_MINT: &str = "disable mint";
+const ACTION_MINT: &str = "mint";
+const ACTION_MINT_TO: &str = "mint to";
+const ACTION_UPDATE_EVENT_INFO: &str = "update event info";
+const ACTION_UPDATE_ADMIN: &str = "update admin";
+const ACTION_UPDATE_MINTER: &str = "update minter";
```

Comment:
I would have a file `action.rs` to store them. And make the value in `snake case` .

Reply:
Yes snake case is better so that they have the same format of `ExecuteMsg`!
Regarding moving to a different file I think it's a bit over separation, no one beside the varius execution functions is using this consts and even if some one else will want to extend this contract probably don't need this strings.